tests/test_async_collections.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+import io
+import os
+import sys
+import time
+import unittest
+from random import randint
+from concurrent.futures import ThreadPoolExecutor
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from gs_chunked_io.async_collections import AsyncSet, AsyncList
+
+
+class TestAsyncSet(unittest.TestCase):
```

Comment:
This doesn't seem to test any set-like features (like inputting a list and having a duplicate removed).  Though from the above, the set-like behavioror  isn't the main focus of the class maybe?  Would it be good to rename these to AsyncPut or AsyncGet?  Or AsyncOrdered and AsyncUnordered?  I'm not wholly sure that those are great either though.

Reply:
I chose "list" and "set" to reflect the ordered and unordered nature of the collections. Since they do not provide a full interface to Python's list and set object, I'll try to think of better names.

Reply:
- Changed `AsyncList` -> `AsyncQueue`.
- Leaving `AsyncSet` because I can't think of anything better.
- Updated docstrings.